onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -7090,6 +7101,21 @@ def aten_scalar_tensor(s: float, dtype: int = FLOAT.dtype) -> RealType:
     return common_ops.cast_to(s, dtype=dtype)
 
 
+@torch_op("aten::scalar_tensor", trace_only=True, complex=True)
+def aten_scalar_tensor_complex(
+    s: Union[FLOAT, DOUBLE], dtype: int = COMPLEX64.dtype
+) -> RealType:
+    """scalar_tensor(Scalar s, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor"""
+    # NOTE: When the input is originally in complex, this function is invoked,
+    # on the other hand, when the input is originally in real, aten_scalar_tensor
+    # is invoked.
+    if dtype == COMPLEX128.dtype:
+        s = op.Cast(s, to=DOUBLE.dtype)
```

Comment:
```suggestion
        result = op.Cast(s, to=DOUBLE.dtype)
```

Reply:
This will cause "Local variable 'result' may be used before it is initialized.", unless we put an else. Do we want to put an else branch with result = op.Identity(s)?

Reply:
Sure, we can put result = s

Reply:
Done with a NOTE